src/poetry/console/commands/add.py

```diff
@@ -266,8 +264,7 @@ def handle(self) -> int:
         self.installer.dry_run(self.option("dry-run"))
         self.installer.verbose(self.io.is_verbose())
         self.installer.update(True)
-        if self.option("lock"):
-            self.installer.lock()
```

Comment:
In another PR I might remove `Installer.lock()` entirely as it's quite confusing. Its behaviour is in fact not in sync with the `--lock` options of the Poetry commands

Reply:
Yeah, quite confusing. IIRC, it's only set for `poetry lock`.

Reply:
It is now ;)